### PR TITLE
allow YAML sources in include_from

### PR DESCRIPTION
**Pull request by korowa**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support configuration substitutions from YAML files

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by rschu1ze** [comment]
@korowa Could you please check if something needs to be changed in the docs, e.g. [here](https://clickhouse.com/docs/en/operations/configuration-files)?

**Comment by korowa** [comment]
> @korowa Could you please check if something needs to be changed in the docs, e.g. [here](https://clickhouse.com/docs/en/operations/configuration-files)?

Since there are no file format limitations mentioned neither in [include_from](https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings#include_from) description, nor in "configuration files" page contents, and doc claims that `server can be configured with configuration files in XML or YAML syntax`, don't think any action required.

Though, I can explicitly specify that `include_from` now supports both syntaxes, if this option is preferred.

**Comment by thevar1able** [comment]
>Though, I can explicitly specify that `include_from` now supports both syntaxes, if this option is preferred.

Yes, please do.

**Comment by korowa** [comment]
@thevar1able Done

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 1]
```diff
@@ -1189,7 +1189,7 @@ Expired time for HSTS in seconds. The default value is 0 means clickhouse disabl
 
 ## include_from {#include_from}
 
-The path to the file with substitutions.
+The path to the file with substitutions. Supported file formats are XML and YAML.
```
Comment (thevar1able) [inline docs/en/operations/server-configuration-parameters/settings.md:1192]:
```suggestion
The path to the file with substitutions. Both XML and YAML formats are supported.
```

**Review by thevar1able — APPROVED** [review]

**Pull request merged**